ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -2625,6 +2722,118 @@ private void verifyDataNodeAndDiskEquivalencyInDc(Datacenter dc, String clusterN
     info("[{}] Successfully verified datanode and disk equivalency in dc {}", dcName.toUpperCase(), dc.getName());
   }
 
+  /**
+   * Fetch IdealStates for all the resource ids.
+   * @param admin The helix admin object.
+   * @param dcName The dc name.
+   * @return
+   */
+  Map<Integer, IdealState> getIdealStateForAllResource(HelixAdmin admin, String dcName) {
+    Map<Integer, IdealState> allIdealStates = new HashMap<>();
+    for (String resourceName : admin.getResourcesInCluster(clusterName)) {
+      if (!resourceName.matches("\\d+")) {
```

Comment:
What does that mean? Is \d some special character for us?

Reply:
it just means that resources has to be a number.